src/lib/flexbox/api/flex-offset.spec.ts

```diff
@@ -0,0 +1,163 @@
+
+    it('should add correct styles for default `fxFlexOffset` usage', () => {
+      componentWithTemplate(`<div fxFlexOffset='32px' fxFlex></div>`);
+      fixture.detectChanges();
+
+      let dom = fixture.debugElement.children[0].nativeElement;
+      let isBox = _.hasStyle(dom, 'margin-left', '32px');
+      let hasFlex = _.hasStyle(dom, 'flex', '1 1 1e-09px') ||         // IE
+          _.hasStyle(dom, 'flex', '1 1 1e-9px') ||          // Chrome
+          _.hasStyle(dom, 'flex', '1 1 0.000000001px') ||   // Safari
+          _.hasStyle(dom, 'flex', '1 1 0px');
+
+      expect(isBox).toBeTruthy();
+      expect(hasFlex).toBeTruthy();
+    });
+
+
+    it('should work with percentage values', () => {
+      expectDOMFrom(`<div fxFlexOffset='17' fxFlex='37'></div>`).toHaveCssStyle({
+        'flex': '1 1 100%',
+        'box-sizing': 'border-box',
+        'margin-left': '17%'
+      });
+    });
+
+    it('should work fxLayout parents', () => {
+      componentWithTemplate(`
+        <div fxLayout='column' class='test'>
+          <div fxFlex='30px' fxFlexOffset='17px'>  </div>
+        </div>
+      `);
+      fixture.detectChanges();
+      let parent = queryFor(fixture, '.test')[0].nativeElement;
+      let element = queryFor(fixture, '[fxFlex]')[0].nativeElement;
+
+      // parent flex-direction found with 'column' with child height styles
+      expect(parent).toHaveCssStyle({'flex-direction': 'column', 'display': 'flex'});
+      expect(element).toHaveCssStyle({'margin-top': '17px'});
+    });
+
+    it('should CSS stylesheet and not inject flex-direction on parent', () => {
+      componentWithTemplate(`
+        <style>
+          .test { flex-direction:column; display: flex; }
+        </style>
+        <div class='test'>
+          <div fxFlexOffset='41px' fxFlex='30px'></div>
+        </div>
+      `);
+
+      fixture.detectChanges();
+      let parent = queryFor(fixture, '.test')[0].nativeElement;
+      let element = queryFor(fixture, '[fxFlex]')[0].nativeElement;
+
+      // parent flex-direction found with 'column' with child height styles
+      expect(parent).toHaveCssStyle({'flex-direction': 'column', 'display': 'flex'});
+      expect(element).toHaveCssStyle({'margin-top': '41px'});
+    });
+
+    it('should work with styled-parent flex directions', () => {
+      componentWithTemplate(`
+        <div fxLayout='row'>
+          <div style='flex-direction:column' class='parent'>
+            <div fxFlex='60px' fxFlexOffset='21'>  </div>
+          </div>
+        </div>
+      `);
+      fixture.detectChanges();
+      let element = queryFor(fixture, '[fxFlex]')[0].nativeElement;
+      let parent = queryFor(fixture, '.parent')[0].nativeElement;
+
+      // parent flex-direction found with 'column'; set child with height styles
+      expect(element).toHaveCssStyle({'margin-top': '21%'});
+      expect(parent).toHaveCssStyle({'flex-direction': 'column'});
+    });
+
+
+    describe('', () => {
```

Comment:
This one's missing a title. If the test inside is standalone, they can be moved outside.